### DISCUSSION TITLE
Launch Azurite programatically

### DISCUSSION THREAD

**Discussion by pelletier197** [Azure/Azurite#2511] [Q&A]
Hello!

I have a use-case where I would like to start Azurite programatically in my tests. I want to start a blob server before my test starts, and close the server at the end of my test. This would largely streamline the test setup for me, and probably to all developers trying to use Azurite with NodeJS. 

It seems like everything is present when installing `azurite` from npm, but the lack of typescript types and documentation makes it pretty much impossible to use as-is. 

Is there a non-documented way to do this? And would it be possible to ship type definitions with the javascript ?

**Comment by blueww** [comment]
Hi @pelletier197 

Azurite don't officially support it, and we have never do any tested it. 
After install Azurite from npm, currently only support start with Azurite command.
